pkg/apis/core/types.go

```diff
@@ -5453,6 +5453,18 @@ type Preconditions struct {
 	UID *types.UID
 }
 
+// LogStreamType represents the desired log stream type.
+type LogStreamType string
+
+const (
+	// LogStreamTypeStdout is the stream type for stdout.
+	LogStreamTypeStdout LogStreamType = "stdout"
```

Comment:
Looking at other places where we decode URL parameters (e.g. `ListOptions.ResourceVersionMatch`) we find a precedent of string-constants being UpperCamelCase, so that alone is not a problem.

The flow, as far as I can tell is:
* `PodLogOptions` is passed up via `kubectl logs`.
* It is then decomposed into a bunch of URL params to call the kubelet API.
* Finally kubelet decodes the params BACK INTO a `PodLogOptions` struct.

My opinion (but consulting with API reviewers) is:

* Constants are UpperCamelCase (Stdout, Stderr) in k8s API
* Do not convert the values' style when building the URL params
* Do not be case-insensitive

As a result, when we decode into a PodLogOptions, the constants are valid and match the k8s API in the first place.

Reply:
Got it. Is this a final decision? I'm not very familiar with the API semantics, so I'll defer to your expertise.

Reply:
Yeah, let's run with "Stdout", "Stderr", and "All" (if you keep it)